Makefile

```diff
@@ -50,3 +50,11 @@ clear-cache:
 	docker rm -vf $$(docker ps -a -q) 2>/dev/null ; \
 	docker rmi -f $$(docker images -a -q) 2>/dev/null ; \
 	docker volume rm $$(docker volume ls -q) 2>/dev/null ; true
+
+experiment=*
+
+stop-trials:
```

Comment:
what's "-r-" and "-d-" in the name? consider avoiding such abbreviations in the code for readability

Reply:
`-r-` is for runners, `-d-` is for dispatchers. Does it make sense?

Reply:
yes, but the suggestion was to avoiding such abbreviations, so the next person don't need to ask this question and/or spend time figuring this out. so could we please use eg `runner` and `dispatcher` instead of `r` and `d`?

Reply:
The reason why we do r- and d- is because there is a relatively short limit on gce instance names since we want to include the experiment name in the instance name we abbreviate so that the experiment name isn't as restricted.

Reply:
It is also the already instance name when creating docker containers. I think this helps to maintain a consistency.

Reply:
> there is a relatively short limit on gce instance names 

how many characters is the limit? do a few characters matter that much?

Reply:
I think you end up with about 50 that can be used for the experiment name. I'd rather use the abbreviations so if we end up with something with a long name "measurer-worker" for example we don't have to be inconsistent or really make the experiment name tight.

Reply:
> The reason why we do r- and d- is because there is a relatively short limit on gce instance names since we want to include the experiment name in the instance name we abbreviate so that the experiment name isn't as restricted.

Don't know this. We can use a full name for dockers' name like runner or dispatcher here.

Reply:
Since the name limit for docker is 128 chars, just to make sure there arrives a consensus: @jonathanmetzman @lszekeres
Plan 1:
- we use `runner-<experiment-name>-<trial_id>` for trial runners containers.
- we use `dispatcher-<experiment-name>` for the dispatcher container.
- we use `measurer-worker-<experiment-name>_<work_id>` for the measurer workers I guess?
Plan 2:
- we use `r-<experiment-name>-<trial_id>` for trial runners containers.
- we use `d-<experiment-name>` for the dispatcher container.
- we use `m-<experiment-name>_<work_id>` for the measurer workers I guess?

Reply:
I'm fine anything, I'm just curious about these limitations and numbers -- I haven't found anything about them.

> I think you end up with about 50 
> Since the name limit for docker is 128 chars

Do you happen to have a reference? I was looking a bunch and I didn't find anything about any length limit on docker container names.

Reply:
The limit is for google cloud instance names since these names are used for them as well. I don't have a reference but you can verify the limit is 63 by trying to create an instance. we also need 3 chars for hyphens and ~7 for for trial ids.

Reply:
Got it, thanks! (For docker i don't think there's a limit: i just tried running `docker run --name <600+ character long name> ...` and it was happy to run it)

Reply:
OK. My fault. I didn't run for a check but just search from Google. [This link](https://docs.docker.com/engine/reference/commandline/tag/#:~:text=A%20name%20component%20may%20not,a%20maximum%20of%20128%20characters.) says a limit for tag, not for name.

Reply:
Let's choose plan 2, which is shorter.